components/script/dom/offscreencanvas.rs

```diff
@@ -142,6 +147,85 @@ impl OffscreenCanvas {
     }
 }
 
+impl Transferable for OffscreenCanvas {
+    type Index = OffscreenCanvasIndex;
+    type Data = TransferableOffscreenCanvas;
+
+    fn can_transfer(&self) -> bool {
+        // TODO (#37882): Allow to transfer with a placeholder canvas element.
+        self.placeholder.is_none()
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#the-offscreencanvas-interface:offscreencanvas-16>
+    fn transfer(&self) -> Result<(OffscreenCanvasId, TransferableOffscreenCanvas), ()> {
+        // Step 1. If value's context mode is not equal to none, then throw an
+        // "InvalidStateError" DOMException.
+        if !self.context.borrow().is_none() {
+            return Err(());
+        }
+
+        // Step 2. Set value's context mode to detached.
+        *self.context.borrow_mut() = Some(OffscreenRenderingContext::Detached);
+
+        // Step 3. Let width and height be the dimensions of value's bitmap.
+        let width = self.width.get();
+        let height = self.height.get();
+
+        // TODO Step 4. Let language and direction be the values of value's
+        // inherited language and inherited direction.
+
+        // Step 5. Unset value's bitmap.
+        self.width.set(0);
+        self.height.set(0);
+
+        // Step 6. Set dataHolder.[[Width]] to width and dataHolder.[[Height]]
+        // to height.
+        // TODO Step 7. Set dataHolder.[[Language]] to language and
+        // dataHolder.[[Direction]] to direction.
+        // TODO Step 8. Set dataHolder.[[PlaceholderCanvas]] to be a weak
+        // reference to value's placeholder canvas element, if value has one, or
+        // null if it does not.
+        let transferred = TransferableOffscreenCanvas { width, height };
+
+        Ok((OffscreenCanvasId::new(), transferred))
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#the-offscreencanvas-interface:transfer-receiving-steps>
+    fn transfer_receive(
+        owner: &GlobalScope,
+        _: OffscreenCanvasId,
+        transferred: TransferableOffscreenCanvas,
+    ) -> Result<DomRoot<Self>, ()> {
+        // Step 1. Initialize value's bitmap to a rectangular array of
+        // transparent black pixels with width given by dataHolder.[[Width]] and
+        // height given by dataHolder.[[Height]].
+        // TODO Step 2. Set value's inherited language to
+        // dataHolder.[[Language]] and its inherited direction to
+        // dataHolder.[[Direction]].
+        // TODO Step 3. If dataHolder.[[PlaceholderCanvas]] is not null, set
+        // value's placeholder canvas element to
+        // dataHolder.[[PlaceholderCanvas]] (while maintaining the weak
+        // reference semantics).
```

Comment:
Here and above, I think it would be good to add some spacing between the various steps for clarity.

Reply:
Ok, added empty lines